crates/worker/src/invoker_integration.rs

```diff
@@ -56,45 +56,48 @@ where
             .map_err(InvocationError::internal)?;
         let request = request_extractor(entry);
 
-        let service_key = match self.key_extractor.extract(
-            &request.service_name,
-            &request.method_name,
-            request.parameter,
-        ) {
-            Ok(k) => k,
-            Err(extraction::Error::NotFound) => {
+        let service_id = match self
+            .schemas
+            .resolve_latest_component_handler(&request.service_name, &request.method_name)
+        {
+            Some(meta) => match meta.ty {
+                ComponentType::Service => ServiceId::unkeyed(request.service_name.clone()),
+                ComponentType::VirtualObject => {
+                    ServiceId::new(request.service_name.clone(), request.key.into_bytes())
+                }
+            },
+            None => {
                 return Err(InvocationError::service_method_not_found(
```

Comment:
I did not rename this because I fear too many conflicts when merging the feature branch. I rather prefer to do these big renamings once we merge the feature branch.